### DISCUSSION TITLE
Alignment of molecules from SMILES

### DISCUSSION THREAD

**Discussion by kellegra** [rdkit/rdkit#6800] [FAQ]
Hi there, 

I am relatively new to RDKit. I would like to rotate a substructure of a set of melucules, so I can align all the other molecules on that rotated substructure. All molecules are provided as smiles. 

My molecules are dipenylethers, so I selected a phenol as substructure. The examples I found for this worked just fine for another set of molecules I tested. But for the diephenylethers I want to have the substrucre oriented differently. 

```python
from rdkit import Chem
from rdkit.Chem import rdDepictor
rdDepictor.SetPreferCoordGen(True)
from rdkit.Chem import rdCoordGen
from rdkit.Chem.Draw import IPythonConsole
from rdkit.Chem import Draw
from rdkit.Chem import rdChemReactions
from rdkit.Chem import AllChem
IPythonConsole.ipython_useSVG=True  #< set this to False if you want PNGs instead of SVGs
IPythonConsole.drawOptions.addStereoAnnotation = True
from rdkit import rdBase
rdBase.rdkitVersion

def mol_with_alignment(mol):
    template = Chem.MolFromSmiles("Oc1ccccc1")
    AllChem.Compute2DCoords(template)
    AllChem.GenerateDepictionMatching2DStructure(mol, template)
    return mol

t = mol_with_alignment(Chem.MolFromSmiles("Oc1ccccc1"))
t
```
![image](https://github.com/rdkit/rdkit/assets/34092761/22be3dcf-c07d-4ad4-9a72-40017d734fe3)

Again, the code  pasted above does its work, but for my purpose I would like to have the molecule rotated, that the hydoxyl group is shown at the top position. Or in other words, the OH group shall be shown at the highlighted position: 

![image](https://github.com/rdkit/rdkit/assets/34092761/c65aef98-4d50-4ca2-8fcd-52c3ff477b6d)

Are there any suggestions, how I could do this? 

Best,
Gerald

**Comment by kienerj** [comment]
Not with SMILES. Your template then must be a drawing in the orientation you prefer that must be conserved when loading.

So you need to use a format that support coordinates most notably:
- CXSMILES
- molfile

If you really need SMILES then CXSMILES is the suggested option. You will probbaly be able to create them, if you don't have Marvin license, by loading a molfile into RDKit and then writing it as cxsmiles  with `MolToCXSmiles`.
